Makefile

```diff
@@ -0,0 +1,25 @@
+##########################################
```

Comment:
I semi-copied this file from the OTEL Collector repo. No need to review it. In a nutshell, it lists all the directories that contain a `Makefile` inside of them, and generates targets to execute them as well as clean targets. For this to properly work, each "child" Makefile should specify what is the default target that will be executed, as the "parent" Makefile relies on just having a single one.

So, for our current situation, we will have the following Make targets generated in the root of the project:
- `terraform`: this will execute the `provision` make target under `terraform/Makefile`.
- `terraform-clean`: this will execute the `clean` make target under `terraform/Makefile`.